cmd/kube-apiserver/app/aggregator.go

```diff
@@ -102,11 +103,23 @@ func createAggregatorServer(aggregatorConfig *aggregatorapiserver.Config, delega
 		autoRegistrationController)
 
 	aggregatorServer.GenericAPIServer.AddPostStartHook("kube-apiserver-autoregistration", func(context genericapiserver.PostStartHookContext) error {
-		go autoRegistrationController.Run(5, context.StopCh)
 		go crdRegistrationController.Run(5, context.StopCh)
+		go func() {
+			// let the CRD controller register APIServices for existing CRDs before starting the autoregistration controller
```

Comment:
this is "what is it doing", not "why is it doing it".  I can tell its waiting on priming for something, but its not obvious why that matters.  Particularly since this won't gate a health check.

Reply:
clarified